Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Pass.cpp

```diff
@@ -113,6 +117,15 @@ namespace AZ
             return desc;
         }
 
+        int Pass::GetDeviceIndex() const
+        {
+            if (m_deviceIndex == AZ::RHI::MultiDevice::InvalidDeviceIndex && m_parent)
+            {
+                return m_parent->GetDeviceIndex();
```

Comment:
If a pass doesn't have m_deviceIndex set, every time GetDeviceIndex is called, it will recursively call it's parent and it's parent all the way up the pass tree.
It would be better to cache the parent device index. We already do something similar with ParentEnable. There's a function called OnHierarchyChange( ) that is invoked any time a pass is added or removed (or it's parent is added or removed), this would be the best place to cache the parent device index to not have to recursively go up the tree.

Reply:
good point, changed as request, please take a look :)